From 672b04fa3d1e4255919ff2fe219d673aade27fb7 Mon Sep 17 00:00:00 2001
From: Nachiket Sahasrabuddhe <nachiket_sahasrabuddhe@xyratex.com>
Date: Wed, 6 Feb 2013 13:02:43 +0530
Subject: [PATCH 2/6] cookie-replacing-senderID

1. packet-ut modified to take into account
   cookie field from sref of item.
---
 rpc/conn.h      | 2 +-
 rpc/slot.c      | 4 ----
 rpc/ut/packet.c | 2 ++
 3 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/rpc/conn.h b/rpc/conn.h
index d8493ca..89e33fd 100644
--- a/rpc/conn.h
+++ b/rpc/conn.h
@@ -251,7 +251,7 @@ enum m0_rpc_conn_flags {
   all the state transitions of conn internally.
  */
 struct m0_rpc_conn {
-	/** Cookie holding a reference to conn on receiver end */
+	/** Cookie holding a reference to conn on other end */
 	struct m0_cookie          c_cookie;
 
 	uint64_t                  c_sender_id;
diff --git a/rpc/slot.c b/rpc/slot.c
index f6511fb..43f46a6 100644
--- a/rpc/slot.c
+++ b/rpc/slot.c
@@ -777,10 +777,6 @@ find_conn(const struct m0_rpc_machine *machine,
 	conn = m0_cookie_of(&sref->sr_ow.osr_cookie,
 			    struct m0_rpc_conn,
 			    c_sender_id);
-#ifndef __KERNEL__
-	if (conn != NULL)
-		printf("\nConn detected using cookie\n");
-#endif
 	if (conn == NULL) {
 		m0_tl_for(rpc_conn, conn_list, conn) {
 			if (m0_rpc_sender_uuid_cmp(&conn->c_uuid,
diff --git a/rpc/ut/packet.c b/rpc/ut/packet.c
index ce50947..72d545f 100644
--- a/rpc/ut/packet.c
+++ b/rpc/ut/packet.c
@@ -22,6 +22,7 @@
 #include "lib/misc.h"
 #include "lib/arith.h"          /* m0_align() */
 #include "lib/ut.h"
+#include "lib/cookie.h"
 #include "mero/init.h"
 #include "mero/magic.h"
 #include "fop/fop.h"
@@ -200,6 +201,7 @@ static void populate_item(struct m0_rpc_item *item)
 	item->ri_slot_refs[0].sr_ow = (struct m0_rpc_onwire_slot_ref) {
 		.osr_uuid.su_uuid = 9876,
 		.osr_session_id = 523,
+		.osr_cookie = cookie_invalid,
 		.osr_slot_id = 23,
 		.osr_verno = {
 			.vn_lsn = 7654,
-- 
1.8.3.2

